eth/client.go

```diff
@@ -287,42 +298,76 @@ func (cl *Client) SubscribeToBlocksStartingFrom(startBlockNumber *big.Int) (*Blo
 	return blockSub, nil
 }
 
-// SubscribeToDeposited writes past Deposited events and newly received ones
+// SubscribeExiting writes past Exiting events and newly received ones
 // into the sink. The `epochs` and `accs` arguments can be used to filter for
 // specific events. Passing `nil` will skip the filtering.
-// Should be started in a go-routine, since it blocks.
-// Can be cancelled via context.
-func (cl *Client) SubscribeToDeposited(ctx context.Context, contract *bindings.Erdstall, epochs []uint64, accs []common.Address, sink chan *bindings.ErdstallDeposited) error {
+// Can be cancelled via Unsubscribe.
+func (cl *Client) SubscribeExiting(ctx context.Context, contract *bindings.Erdstall, epochs []uint64, accs []common.Address) (*ExitingSubscription, error) {
 	// sub to new events
 	wOpts, err := cl.NewWatchOpts(ctx)
 	if err != nil {
-		return err
+		return nil, err
 	}
-	sub, err := contract.WatchDeposited(wOpts, sink, epochs, accs)
+	start := uint64(0) // Begin the query at block 0.
```

Comment:
And even if you want to subscribe to _all_ events, you can still start at `InitBlock`, no need to start at 0, which would be quite unnecessary on a testnet or mainnet.

Reply:
The `InitBlock` should be used here indeed, but i dont know if we even need past events.  
I guess in most cases it does not matter, only in some corner cases when the client starts and the contract is already frozen or something.  
The `eth` client does not know the last block number, so it would have to calculate the first blocknumber that it should use for the query from the `epochs`  argument.

Reply:
I added a `startBlock *uint64` argument to both subscribe functions.

Reply:
We discussed with @matthiasgeihs that subscriptions only for for future blocks anyways. So we need to check if it's sufficient in our use cases to only have subscriptions to events starting at the moment of subscription. We can then remove the `startBlock` because we don't need to only start at some future block.

Reply:
> Please check the usage of our subscriptions, if they work with normal subs, without past filters. If yes, we can remove the `startBlock`. If no, we have to additionally do filtering into the past. Then also check, if we really want all events since `InitBlock` or actually only of the active epoch.

I tried it three times with three clients both malicious and honest case and it worked. Maybe we could even do a CI test at some point :raised_hands: